dunai/src/Data/MonadicStreamFunction/InternalCore.hs

```diff
@@ -133,3 +133,8 @@ reactimate :: Monad m => MSF m () () -> m ()
 reactimate sf = do
   (_, sf') <- unMSF sf ()
   reactimate sf'
+
+vain :: Monad m => MSF m a b -> MSF m a (b, MSF m a b)
+vain msf0 = MSF { unMSF = f }
```

Comment:
Can you format this in the same way you formatted `link`?

Reply:
Done.